site/content/blog/2025-02-17-portswigger-lab-walkthrough-broken-brute-force-protection-ip-block/index.md

```diff
@@ -0,0 +1,793 @@
+> 
+> - Your credentials: `wiener:peter`
+> - Victim's username: `carlos`
+> - [Candidate passwords](https://portswigger.net/web-security/authentication/auth-lab-passwords)
+
+Brute-force protection is especially important for applications that depend entirely on password-based authentication. Without additional guard rails like multi-factor authentication (MFA), an attacker who can make multiple or unlimited guesses of users’ login credentials could eventually gain access to their accounts.
+
+However, poor implementation can undermine even the best defense mechanisms. If developers make incorrect assumptions about how users will interact with their application, they may unintentionally introduce vulnerabilities that attackers can exploit.
+
+This can be in the form of a logic flaw — a vulnerability that allows users to manipulate normal application functionality or workflow to achieve results the developer neither intended nor expected.
+
+In this lab, we’ll exploit a logic flaw to access another user’s account. The lab provides us with a specific username, “carlos”. So, we don’t need to guess a valid username — we can focus on brute-forcing the password.
+
+Visit [the lab page](https://portswigger.net/web-security/authentication/password-based/lab-broken-bruteforce-protection-ip-block) and click “Access the Lab” to launch the test site. The URL for the lab will look something like “[https://0a19000b0319d4b784b7a40400f800ed.web-security-academy.net/](https://0a19000b0319d4b784b7a40400f800ed.web-security-academy.net/)”. You’ll want to copy it.
+
+For eagle-eyed readers, you may notice that the URLs in the screenshots vary. This is because multiple lab instances were used to write this article:
+
+&nbsp;
+![lab-home.png](images/lab-home.png)
+&nbsp;
+
+Now, let’s open the lab URL you just copied via ZAP. Click *Manual Explore* in the *Quick Start* tab:
+
+&nbsp;
+![quick-start-manual-explore-screen.png](images/quick-start-manual-explore-screen.png)
+&nbsp;
+
+Copy-paste the lab URL into the *URL to explore* field and launch your browser with the *Launch Browser* button. Or, use the browser icon in the main toolbar:
+
+&nbsp;
+![launch-test-site-ZAP-configured-browser.png](images/launch-test-site-ZAP-configured-browser.png)
+&nbsp;
+
+> [!NOTE]
+> If you have any issues launching your browser from ZAP, see the following pages for help. ZAP uses add-ons to enhance its core features, and one of these is Selenium. This add-on provides drivers for interfacing with supported browsers. It is installed by default and exposes some configurable options:
+>
+> 1. [How can I fix 'browser was not found'?](https://www.zaproxy.org/faq/how-can-i-fix-browser-was-not-found/) - Zaproxy Docs (FAQ)
+> 2. [Manual Explore](https://www.zaproxy.org/docs/desktop/addons/quick-start/#manual-explore) - Zaproxy Docs
+> 3. [Selenium](https://www.zaproxy.org/docs/desktop/addons/selenium/) - Zaproxy Docs
```

Comment:
I'd suggest "ZAP" vs "Zaproxy" for the hyphen notes..... though maybe those aren't necessary here either? 🤷‍♂️